pmemkv-binding/src/main/java/io/pmem/pmemkv/Database.java

```diff
@@ -30,15 +31,41 @@
 	Converter<K> keyConverter;
 	Converter<V> valueConverter;
 
-	private ByteBuffer getDirectBuffer(ByteBuffer buf) {
+	private static class ThreadDirectBuffers {
+		private final ArrayList<ThreadLocal<ByteBuffer>> buffers = new ArrayList<>(2);
+
+		public ThreadDirectBuffers() {
+			for (int i = 0; i < 2; ++i) {
+				buffers.add(new ThreadLocal<ByteBuffer>() {
+					@Override
+					protected ByteBuffer initialValue() {
+						return ByteBuffer.allocateDirect(10000); // max size of key/value
```

Comment:
are we going to make this limit some variable?
the default usage will change...?